### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by MaxSchmide**

### CODE REVIEW

File: src/formatDate.js [hunk 1]
```diff
@@ -50,7 +50,52 @@
...
+        }
+
+        break;
+      case 'M':
+        newDateArr.push(currentMonth);
+        break;
+      case 'D':
+        newDateArr.push(currentDay);
+        break;
+    }
```
Comment (ilona-borkovska) [inline src/formatDate.js:95]: image

**Review by ilona-borkovska — APPROVED** [review]

File: src/formatDate.js [hunk 2]
```diff
@@ -50,7 +50,56 @@
...
-  // write code here
+  const oldSeparator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+  let currentYear = '';
+  let currentMonth = '';
+  let currentDay = '';
+  const oldDateArr = date.split(oldSeparator);
+  const newDateArr = [];
+
+  fromFormat.map((el, index) => {
```
Comment (anstsot) [inline src/formatDate.js:61]: `map` method should always return value for each iteration. here would be better to use `forEach`

File: src/formatDate.js [hunk 3]
```diff
@@ -50,7 +50,56 @@
...
+        break;
+      case 'D':
+        currentDay = oldDateArr[index];
+        break;
+      default:
+        throw new Error('Invalid format');
+    }
+  });
+
+  toFormat.map((el) => {
```
Comment (anstsot) [inline src/formatDate.js:77]: same here. use `forEach` or consider using `for ... of`

File: src/formatDate.js [hunk 4]
```diff
@@ -50,7 +50,56 @@
...
+      case ('Y'):
+        if (el.length > currentYear.length) {
+          const year = (+currentYear < 30 ? '20' : '19') + currentYear;
+
+          newDateArr.push(year);
+        } else if (el.length < currentYear.length) {
+          newDateArr.push(currentYear.slice(-2));
+        } else {
+          newDateArr.push(currentYear);
+        }
```
Comment (anstsot) [inline src/formatDate.js:88]: you can modify `currentYear` variable in `if ... else` constructions. and after that push it to `newDateArr`

**Review by anstsot — CHANGES_REQUESTED** [review]
Almost done :+1:
Let’s add a few fixes to improve your code.

**Review by anstsot — APPROVED** [review]
Good work, approving:sparkles:

**Pull request closed without merging**